src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -217,7 +220,7 @@ private void OnSerializing(StreamingContext ctx)
         ///  what happens for a version update)
         /// </summary>
 
-        public virtual string Name
+        public virtual string? Name
```

Comment:
_sortName is not nullable and we debug.Assert `m_name` not being `null` -- are you making this one nullable because it is a virtual call? Does overriden members return null?

cc: @stephentoub @terrajobst another case

Reply:
the reason I made `m_name` nullable is because it becomes very briefly null during deserialization :[ that destroys basically whole annotating - technically speaking it shouldn't be null but the way it works it must be null before serialization is started so that we can check if serialization set the field

Reply:
Can’t we return `m_name!` ?

Reply:
I was able to make m_name non-nullable even though it is nullable for a brief moment - it didn't make sense to me to mark it as nullable only because of deserialization scenario